### PR TITLE
Add option to use deployment_name instead of workspace string

### PR DESCRIPTION
**Pull request by arbulu89**
Implement the option to create the infrastructure names based on a new variable named `deployment_name`. This variable has preference over the terraform workspace, but if it's not used, the terraform workspace is used, as it is in the current code. So, there is not any non backward compatible change

### CODE REVIEW

**Review by MalloZup — COMMENTED** [review]
I/m ok with this.

I would however rename it to more strong from `deployment_name`  to `deployment_unique_name` or `deployment_uid` to make clear that the name is a UID if the user changes it, or do with 2 same there are problems.

**Review by MalloZup — CHANGES_REQUESTED** [review]
I'm ok with this! 

 but I would rename the variable with `deployment_unique_name` or `deployment_UID` or something more strong the `deployment_name`

**Comment by arbulu89** [comment]
I don't know if changing names would clarify that much. Maybe adding this information in the `description` of the variable and the examples file would be enough.

The variable is used to give a name to the deployment, the fact that it must be unique is more a feature/limitation than a description.

**Review by MalloZup — APPROVED** [review]
I'm personally like the `unique` but I'm ok if we add description

File: aws/variables.tf [hunk 1]
```diff
@@ -65,9 +65,10 @@ variable "private_key_location" {
 
 # Deployment variables
 
-variable "name" {
-  description = "hostname, without the domain part"
+variable "deployment_name" {
+  description = "Name used to complement some of the infrastructure resources name as sufix. If it is not provided, the terraform workspace string is used"
```
[the remark this answers is not in the document]
Comment (Simranpal) [reply aws/variables.tf]: Ok i see you correct the typo

[the remark this answers is not in the document]
Comment (arbulu89) [reply aws/variables.tf]: Thanks @Simranpal 
Updated to: `Suffix string added to some of the infrastructure resources names. If it is not provided, the terraform workspace string is used as suffix`

As not all of the infrastructure resources have the suffix

**Pull request merged**